experimental/origin-dapp2/src/pages/user/User.js

```diff
@@ -1,13 +1,24 @@
 import React, { Component } from 'react'
 import { Query } from 'react-apollo'
 import get from 'lodash/get'
+import { fbt } from 'fbt-runtime'
 
 import IdentityQuery from 'queries/Identity'
 import Reviews from 'components/Reviews'
 import Avatar from 'components/Avatar'
 
 import UserListings from './_UserListings'
 
+function parseUserSocialID(attestations) {
+  const socialUrlObject = {}
+  attestations.forEach(attestation => {
+    const socialData = JSON.parse(attestation)
+    socialUrlObject[socialData.data.attestation.site.siteName] =
+      socialData.data.attestation.site.userId.raw
```

Comment:
A few comments:
 - attestation.site is only valid for social networks. not for phone and email attestations.
 - even if attestation.site exists, it is not guaranteed attestation.site.userId is (it is only currently only populated for Twitter and Airbnb. not for Facebook for example).
 - finally if attestation.site.userId, raw field may not be presented (id may be stored encrypted).

As a result, in some cases an exception will get thrown in the forEach loop and cause its execution to stop, which will cause the socialUrlObject to not be fully populated...

I'd suggest checking for the presence of the fields before dereferencing them.
e.g. something like this:
if (socialData.data.attestation.site && socialData.data.attestation.site.userId && socialData.data.attestation.site.userId) {
   socialUrlObject[socialData.data.attestation.site.siteName] = socialData.data.attestation.site.userId.raw
}

Reply:
Will do, thanks for the feedback! I'll make sure to condition out the loop in case any new social media comes through, I'll see if I can figure out a way to do it without having to hard code the references to .site.useId for example. That was it won't need to be touched in the future.